crates/trigger/src/loader.rs

```diff
@@ -13,16 +11,46 @@ use tokio::fs;
 
 use spin_common::{ui::quoted_path, url::parse_file_url};
 
+/// Compilation status of all components of a Spin application
+pub enum CompilationStatus {
+    #[cfg(feature = "aot-compilation")]
+    /// All components are componentized and ahead of time (AOT) compiled to cwasm.
+    AllAotComponents,
+    /// No components are ahead of time (AOT) compiled.
+    NoneAot,
+}
+
+/// Loader for the Spin runtime
 pub struct TriggerLoader {
+    /// Working directory where files for mounting exist.
     working_dir: PathBuf,
+    /// Set the static assets of the components in the temporary directory as writable.
     allow_transient_write: bool,
+    /// Declares the compilation status of all components of a Spin application.
+    compilation_status: CompilationStatus,
 }
 
 impl TriggerLoader {
     pub fn new(working_dir: impl Into<PathBuf>, allow_transient_write: bool) -> Self {
         Self {
             working_dir: working_dir.into(),
             allow_transient_write,
+            compilation_status: CompilationStatus::NoneAot,
+        }
+    }
+
+    /// Creates a TriggerLoader that loads AOT compiled components. It enables
+    /// uses the unsafe `wasmtime::deserialize` function, which is only safe if
+    /// used on components serialized with `Wasmtime Module::serialize` and
+    /// `Engine::precompile_module`. See the Wasmtime documentation for more
+    /// information:
+    /// https://docs.rs/wasmtime/latest/wasmtime/struct.Module.html#method.deserialize
+    #[cfg(feature = "aot-compilation")]
+    pub fn new_aot_compiled(working_dir: impl Into<PathBuf>, allow_transient_write: bool) -> Self {
```

Comment:
I do think we should mark this as `unsafe`. It doesn't immediately do anything unsafe itself but its returned value has potentially unsafe behaviors. It is roughly equivalent to some wasmtime config options like https://docs.rs/wasmtime/18.0.2/wasmtime/struct.Config.html#method.cranelift_flag_enable

Reply:
OoOo TIL. That's a good catch.

Reply:
Thats a great example. I changed this into a `unsafe` setter that sets `CompilationStatus::AllAotComponents` instead of a constructor.